utils/reader.go

```diff
@@ -347,14 +347,22 @@ func (reader *Reader) IsEmpty() bool {
 	return reader.offset >= uint64(len(reader.buf))
 }
 
-// RemainingBytes returns the number of bytes that have not yet been read.
-func (reader *Reader) RemainingBytes() uint64 {
+// RemainingBytesLen returns the number of bytes that have not yet been read.
+func (reader *Reader) RemainingBytesLen() uint64 {
 	if reader.IsEmpty() {
 		return 0
 	}
 	return uint64(len(reader.buf)) - reader.offset
 }
 
+// RemainingBytes returns the bytes that have not yet been read.
+func (reader *Reader) RemainingBytes() []byte {
```

Comment:
We should remove this function. I intentionally left out a way to grab some portion of the buffer to enforce the use of the tested reader functions.